src/lib.rs

```diff
@@ -1834,7 +1845,12 @@ impl Drop for BufferViewMut<'_> {
 impl Buffer {
     /// Return the binding view of the entire buffer.
     pub fn as_entire_binding(&self) -> BindingResource {
-        BindingResource::Buffer {
+        BindingResource::Buffer(self.as_entire_buffer_binding())
+    }
+
+    /// Return the binding view of the entire buffer.
+    pub fn as_entire_buffer_binding(&self) -> BufferBinding {
```

Comment:
Can be merged with `as_entire_binding` if we don't mind breaking changes.

Reply:
We don't mind, but also it seems useful to have this one now.
We can either keep the old one, or even do `impl From<BufferBinding> for BindingResource {..}`

Reply:
`From` is definitely more idiomatic, but is harder to discover in docs so we should call it out specifically. I have received a suggestion to have From<> for BindingResource for each type of binding.

Reply:
I'm concerned about deriving `From<>` for references though. I believe this is a bad practice, since at the call site you'd see `xxx.into()` and assume it's converted by consumption, while it's not.

Reply:
Hmm, yeah it also won't auto-convert for you so if you have a Vec<BufferBinding> you need to call `xxx.as_ref().into()` which is kinda lame. I definitely feel like we should have a consistent abstraction here somehow, but maybe that's just making every resource have an `as_binding()`...

Reply:
Yeah, that doesn't sound too bad.